aws/resource_aws_fms_policy.go

```diff
@@ -138,31 +145,7 @@ func resourceAwsFmsPolicy() *schema.Resource {
 func resourceAwsFmsPolicyCreate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).fmsconn
 
-	fmsPolicy := &fms.Policy{
-		PolicyName:          aws.String(d.Get("name").(string)),
-		RemediationEnabled:  aws.Bool(d.Get("remediation_enabled").(bool)),
-		ResourceType:        aws.String("ResourceTypeList"),
-		ResourceTypeList:    expandStringSet(d.Get("resource_type_list").(*schema.Set)),
-		ExcludeResourceTags: aws.Bool(d.Get("exclude_resource_tags").(bool)),
-	}
-
-	securityServicePolicy := d.Get("security_service_policy_data").([]interface{})[0].(map[string]interface{})
-	fmsPolicy.SecurityServicePolicyData = &fms.SecurityServicePolicyData{
-		ManagedServiceData: aws.String(securityServicePolicy["managed_service_data"].(string)),
-		Type:               aws.String(securityServicePolicy["type"].(string)),
-	}
-
-	if rTags, tagsOk := d.GetOk("resource_tags"); tagsOk {
-		fmsPolicy.ResourceTags = constructResourceTags(rTags)
-	}
-
-	if v, ok := d.GetOk("include_map"); ok {
-		fmsPolicy.IncludeMap = expandFMSPolicyMap(v.([]interface{}))
-	}
-
-	if v, ok := d.GetOk("exclude_map"); ok {
-		fmsPolicy.ExcludeMap = expandFMSPolicyMap(v.([]interface{}))
-	}
+	fmsPolicy := resourceAwsFmsPolicyExpandPolicy(d)
```

Comment:
I think this change is breaking creation because the resource ID and PolicyUpdateToken are not yet available:

```
=== CONT  TestAccAWSFmsPolicy_tags
    resource_aws_fms_policy_test.go:129: Step 1/2 error: Error running apply:
        Error: Creating Policy Failed: InvalidParameter: 2 validation error(s) found.
        - minimum field size of 36, PutPolicyInput.Policy.PolicyId.
        - minimum field size of 1, PutPolicyInput.Policy.PolicyUpdateToken.



=== CONT  TestAccAWSFmsPolicy_cloudfrontDistribution
    resource_aws_fms_policy_test.go:47: Step 1/2 error: Error running apply:
        Error: Creating Policy Failed: InvalidParameter: 2 validation error(s) found.
        - minimum field size of 36, PutPolicyInput.Policy.PolicyId.
        - minimum field size of 1, PutPolicyInput.Policy.PolicyUpdateToken.



=== CONT  TestAccAWSFmsPolicy_basic
    resource_aws_fms_policy_test.go:19: Step 1/2 error: Error running apply:
        Error: Creating Policy Failed: InvalidParameter: 2 validation error(s) found.
        - minimum field size of 36, PutPolicyInput.Policy.PolicyId.
        - minimum field size of 1, PutPolicyInput.Policy.PolicyUpdateToken.



=== CONT  TestAccAWSFmsPolicy_includeMap
    resource_aws_fms_policy_test.go:75: Step 1/2 error: Error running apply:
        Error: Creating Policy Failed: InvalidParameter: 2 validation error(s) found.
        - minimum field size of 36, PutPolicyInput.Policy.PolicyId.
        - minimum field size of 1, PutPolicyInput.Policy.PolicyUpdateToken.



--- FAIL: TestAccAWSFmsPolicy_tags (11.14s)
--- FAIL: TestAccAWSFmsPolicy_cloudfrontDistribution (13.52s)
=== CONT  TestAccAWSFmsPolicy_update
    resource_aws_fms_policy_test.go:104: Step 1/2 error: Error running apply:
        Error: Creating Policy Failed: InvalidParameter: 2 validation error(s) found.
        - minimum field size of 36, PutPolicyInput.Policy.PolicyId.
        - minimum field size of 1, PutPolicyInput.Policy.PolicyUpdateToken.



--- FAIL: TestAccAWSFmsPolicy_basic (14.61s)
--- FAIL: TestAccAWSFmsPolicy_includeMap (15.72s)
--- FAIL: TestAccAWSFmsPolicy_update (16.84s)
```